test/plugin/test_out_elasticsearch.rb

```diff
@@ -53,6 +53,10 @@ def stub_elastic_with_store_index_command_counts(url="http://localhost:9200/_bul
     end
   end
 
+  def stub_elastic_bulk_error(url="http://localhost:9200/_bulk")
+    stub_request(:post, url).to_return(:status => 200, :body => '{"took":6,"errors":true,"items":[{"index":{"_index":"testing-bulk-indexing-0","_type":"type-00","_id":"1","_version":1,"_shards":{"total":1,"successful":1,"failed":0},"status":201}},{"index":{"_index":"testing-bulk-indexing-1","_type":"type-00","_id":"1","status":400,"error":{"type":"mapper_parsing_exception","reason":"failed to parse [field2]","caused_by":{"type":"illegal_argument_exception","reason":"Invalid format: \"val-field2-00\""}}}},{"index":{"_index":"testing-bulk-indexing-2","_type":"type-00","_id":"1","_version":1,"_shards":{"total":1,"successful":1,"failed":0},"status":201}}]}', :headers => {})
```

Comment:
you don't have to use webmock to mock the low level POST and response
you could just mock out the client.bulk call directly
something like this - where `ClientClass` is the class of the `client` object in 
```
client = mock
stub(Elasticsearch::Client).new { client }
stub(client).bulk { whatever you want to return here }
```